core/src/main/scala/org/apache/spark/deploy/ApplicationDescription.scala

```diff
@@ -28,5 +28,14 @@ private[spark] class ApplicationDescription(
 
   val user = System.getProperty("user.name", "<unknown>")
 
+  def copy(
```

Comment:
Is there a reason why we shouldn't just make `ApplicationDescription` into a case class, since that will implicitly give us a `copy` method?